src/lseq/mod.rs

```diff
@@ -127,20 +127,24 @@ impl LSeq {
 
     }
 
+    /// Get the length of the LSEQ
+    pub fn len(&self) -> usize {
+        self.seq.len()
+    }
     /// Get the string represented by the LSeq tree.
-    pub fn text(&self) -> String {
-        self.text.iter().map(|(_, _, _, c,)| c).collect::<String>()
+    pub fn sequence(&self) -> impl Iterator<Item = T> + '_ {
```

Comment:
The name `iter` would be more idiomatic, I think. But maybe @davidrusu could weigh in being the library author